compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6356,26 +6358,38 @@ private BType getCompletionType(BType collectionType, BLangQueryExpr queryExpr,
         return completionType;
     }
 
-    private BType getNonContextualQueryType(BType staticType, BType basicType) {
-        BType resultType;
+    private BType getNonContextualQueryType(BType staticType, BType basicType, Location pos) {
+        basicType = Types.getReferredType(basicType);
         switch (basicType.tag) {
             case TypeTags.TABLE:
-                resultType = symTable.tableType;
+                if (types.isAssignable(staticType, symTable.mapAllType)) {
+                    return symTable.tableType;
+                }
                 break;
             case TypeTags.STREAM:
-                resultType = symTable.streamType;
-                break;
+                return symTable.streamType;
```

Comment:
Add a test for stream constrained with literal

Reply:
Updated. 933052368982fe42d170de9491275b990a8ce1ed